runtime/src/bank.rs

```diff
@@ -3590,51 +3630,6 @@ impl Bank {
         }
     }
 
-    pub fn compare_bank(&self, dbank: &Bank) {
-        if ptr::eq(self, dbank) {
-            return;
-        }
```

Comment:
maybe this should be kept to deadlock free? #10466 

Yeah, we should have added an comment...

Reply:
ah yeah good catch, thanks!